spec.md

```diff
@@ -516,6 +539,37 @@ The type of the submodule instance `passthrough`{.firrtl} above is thus:
 { flip in : UInt<8>, out : UInt<8> }
 ```
 
+#### Instance Choices
+
+FIRRTL supports the specialization of designs by allowing instances to target a module chosen based on an `option`{.firrtl}.
+
+Example:
+
+``` firrtl
+circuit:
+  option Platform:
+    case FPGA:
+    case ASIC:
+
+  module InstanceChoice:
+    inst_choice clock_gate of DefaultTarget, Platform:
+      FPGATarget => FPGA
+
+  module DefaultTarget:
+
+  module FPGATarget:
+```
+
+The instance choice declaration specifies the instance name and names the option group based on which the choices are selected.
+It introduces mappings with the `when`{.firrtl} keyword, attaching a module to a specific case allowed by the option group.
+A default option must always be provided, to be implicitly used when no specialisation is requested at any stage.
+The mapping does not need to be exhaustive, cases can be left unspecified, resorting to the default mapping.
+Only modules or extmodules can be targeted.
+
+Probes or property ports are not allowed on modules which are part of instance choices.
```

Comment:
Why properties given that these are "evaluated" later?

Why no probes? It seems like this could be made to work if the `define` for a reference included another define that the optional instance must define?

Reply:
They can be made to work, but it's not clear to me how at the moment and the feature is useful even without them.